opentech/apply/users/views.py

```diff
@@ -20,8 +25,62 @@ def account(request):
 @login_required(login_url=reverse_lazy('users:login'))
 @require_oauth_whitelist
 def oauth(request):
+    """Generic, empty view for the OAuth associations."""
+
+    return TemplateResponse(request, 'users/oauth.html', {})
+
+
+class ActivationView(TemplateView):
     """
-    Generic, empty view for the OAuth associations.
+    Inspired by https://github.com/ubernostrum/django-registration
     """
 
-    return TemplateResponse(request, 'users/oauth.html', {})
+    def get(self, request, *args, **kwargs):
+        user = self.activate(*args, **kwargs)
```

Comment:
Not too happy with the way this reads. Would prefer something like below which means you don't need to deconstruct the class to understand.

```
user = self.get_user(uidb64)
if self.valid(user, token):
    self.activate(user)
    login(user)
    return redirect()
return redirect('invalid')
```

Reply:
👍